calico/felix/endpoint.py

```diff
@@ -59,13 +65,29 @@ def __init__(self, config, ip_type,
 
         # All endpoint dicts that are on this host.
         self.endpoints_by_id = {}
-        # Dict that maps from interface name ("tap1234") to endpoint ID.
+        # Dict that maps from int`erface name ("tap1234") to endpoint ID.
         self.endpoint_id_by_iface_name = {}
 
         # Set of endpoints that are live on this host.  I.e. ones that we've
         # increffed.
         self.local_endpoint_ids = set()
 
+        # Index tracking what policy applies to what endpoints.
+        self.policy_index = LabelValueIndex()
+        self.policy_index.on_match_started = self.on_policy_match_started
+        self.policy_index.on_match_stopped = self.on_policy_match_stopped
+        self._label_inherit_idx = LabelInheritanceIndex(self.policy_index)
+        # Tier orders by tier ID.  Tiers with lower orders are applied first.
```

Comment:
Really 'by tier ID'?  How does that help with ordering, then?  Perhaps an example here would help.